spec/factories/collection_template.rb

```diff
@@ -0,0 +1,831 @@
+require 'faker'
+
+FactoryGirl.define do
+  factory :full_collection_template, class: CollectionTemplate do
+    transient do
+      draft_short_name nil
+      draft_entry_title nil
+      version nil
+      collection_data_type nil
+    end
+
+    provider_id 'MMT_2'
+    draft_type 'CollectionTemplate'
+
+    trait :with_valid_dates do
+      draft {{
+        'MetadataDates' => [{
+          'Type' => 'CREATE',
+          'Date' => '2010-12-25T00:00:00Z'
+        }, {
+          'Type' => 'UPDATE',
+          'Date' => '2010-12-30T00:00:00Z'
+        }]
+      }}
+    end
+
+    draft {{
```

Comment:
I wonder if there is a way to save this draft metadata and have it be used by multiple factory methods, since it is fairly long and we might also want to use it for draft proposals.